symengine/polys/mpoly_symengine.h

```diff
@@ -412,48 +406,50 @@ class MSymEnginePoly : public Basic
         return Container(std::move(d), numeric_cast<unsigned>(s.size()));
     }
 
-    inline vec_basic get_args() const
-    {
-        return {};
-    }
-
-    inline const Container &get_poly() const
-    {
-        return poly_;
-    }
-
-    inline const set_basic &get_vars() const
-    {
-        return vars_;
-    }
-
     bool __eq__(const Basic &o) const
     {
         // TODO : fix for when vars are different, but there is an intersection
         if (not is_a<Poly>(o))
             return false;
         const Poly &o_ = down_cast<const Poly &>(o);
         // compare constants without regards to vars
-        if (1 == poly_.dict_.size() && 1 == o_.poly_.dict_.size()) {
-            if (poly_.dict_.begin()->second != o_.poly_.dict_.begin()->second)
+        if (1 == this->get_poly().dict_.size()
+            && 1 == o_.get_poly().dict_.size()) {
+            if (this->get_poly().dict_.begin()->second
+                != o_.get_poly().dict_.begin()->second)
                 return false;
-            if (poly_.dict_.begin()->first == o_.poly_.dict_.begin()->first
-                && unified_eq(vars_, o_.vars_))
+            if (this->get_poly().dict_.begin()->first
+                    == o_.get_poly().dict_.begin()->first
+                && unified_eq(this->get_vars(), o_.get_vars()))
                 return true;
             typename Container::vec_type v1, v2;
-            v1.resize(vars_.size(), 0);
-            v2.resize(o_.vars_.size(), 0);
-            if (poly_.dict_.begin()->first == v1
-                || o_.poly_.dict_.begin()->first == v2)
+            v1.resize(this->get_vars().size(), 0);
+            v2.resize(o_.get_vars().size(), 0);
+            if (this->get_poly().dict_.begin()->first == v1
+                || o_.get_poly().dict_.begin()->first == v2)
                 return true;
             return false;
-        } else if (0 == poly_.dict_.size() && 0 == o_.poly_.dict_.size()) {
+        } else if (0 == this->get_poly().dict_.size()
+                   && 0 == o_.get_poly().dict_.size()) {
             return true;
         } else {
-            return (unified_eq(vars_, o_.vars_)
-                    && unified_eq(poly_.dict_, o_.poly_.dict_));
+            return (unified_eq(this->get_vars(), o_.get_vars())
+                    && unified_eq(this->get_poly().dict_, o_.get_poly().dict_));
         }
     }
+
+    static set_basic get_translated_container(Container &x, Container &y,
```

Comment:
Was moved inside the class, as different classes can have different `get_translated_container` methods, which serve different purposes. Hence the `add/sub/mul/div` functions use that particular class's `get_translated` method, and remain the same for all classes. (just how `+, -, x, /` were overloaded for univariate poly classes and implemented commonly for all classes)